handler/authorize.go

```diff
@@ -0,0 +1,78 @@
+package handler
+
+import (
+	"encoding/json"
+	"net/http"
+
+	"github.com/gitpods/gitpod"
+	"github.com/gitpods/gitpod/store"
+	"github.com/go-kit/kit/log"
+	"github.com/go-kit/kit/log/level"
+	"golang.org/x/crypto/bcrypt"
+)
+
+const (
+	loginAttemptFailed  = "failed"
+	loginAttemptSuccess = "success"
+)
+
+var (
+	BadCredentialsJson = map[string]string{"message": "Bad credentials"}
+)
+
+type LoginStore interface {
+	GetUserByEmail(string) (gitpod.User, error)
+}
+
+type loginForm struct {
+	Email    string `json:"email"`
+	Password string `json:"password"`
+}
+
+func Authorize(logger log.Logger, s LoginStore) http.HandlerFunc {
+	// TODO: This registers this collector globally, which causes them to register twice in tests and thus fails.
```

Comment:
Yeah, because the metrics are only registered in `package main`, you never pass a prometheus counter around within your tests, but rather a https://godoc.org/github.com/go-kit/kit/metrics/discard#NewCounter